### PR TITLE
Fix #227

### LINKED ISSUE #227: [API] Query ERC20 contracts
`/tokens/list?query=OMG`

Server-side we should be looking for smart contract name, symbol or address.

### PR DESCRIPTION
**Pull request by hewigovens**
1. add token search api
2. add name / symbol index
3. fix readOneToken parameter

### CODE REVIEW

File: src/models/Erc20ContractModel.ts [hunk 1]
```diff
@@ -27,5 +27,7 @@ const ERC20ContractScheme = new Schema({
 
 // indices
 ERC20ContractScheme.index({address: 1}, {name: "contractAddressIndex"});
+ERC20ContractScheme.index({symbol: 1}, {name: "contractSymbolIndex"});
+ERC20ContractScheme.index({name: 1}, {name: "contractNameIndex"});
```
Comment (vikmeup) [inline src/models/Erc20ContractModel.ts]: No need to specify name tho, it will name it automatically

File: src/controllers/TokenController.ts [hunk 2]
```diff
@@ -79,20 +80,19 @@ export class TokenController {
...
     public readOneToken(req: Request, res: Response) {
-        if (!req.params || !req.params.address) {
+        if (!req.params || !req.params.tokenWalletAddress) {
             sendJSONresponse(res, 404, { "message": "No address in request" });
             return;
         }
-
         // validate wallet address
-        req.checkParams("address", "wallet address must be alphanumeric").isAlphanumeric();
+        req.checkParams("tokenWalletAddress", "wallet address must be alphanumeric").isAlphanumeric();
```
Comment (vikmeup) [inline src/controllers/TokenController.ts]: Why is this changed to `tokenWalletAddress`?

Comment (hewigovens) [reply src/controllers/TokenController.ts]: match url routing

Comment (vikmeup) [reply src/controllers/TokenController.ts]: What does it mean? `api.trustwalletapp.com/tokens?address=0xe47494379c1d48ee73454c251a6395fdd4f9eb43` this is current endpoint that's being used to retrieve tokens

Comment (hewigovens) [reply src/controllers/TokenController.ts]: I’m not sure, it’s old logic, I just fix the wrong premeter name.

Comment (vikmeup) [reply src/controllers/TokenController.ts]: can you revert that name change? I think it will break current logic

**Comment by vikmeup** [comment]
@hewigovens can you rebase against master and fix tests? let's get this merged!

**Comment by hewigovens** [comment]
@vikmeup done~
